merge/merge.go

```diff
@@ -0,0 +1,171 @@
+package diff
+
+import (
+	"log"
+	"strconv"
+)
+
+const indicesReorderedKey = "$"
+
+// Merge merges an update diff into the previous version of a JSON object,
+// and returns the updated version.
+//
+// It effectively reverses the diff algorithm implemented in package diff.
+func Merge(prev map[string]interface{}, diff map[string]interface{}) interface{} {
+	new := make(map[string]interface{})
+
+	// Update exisiting fields.
+	for k, v := range prev {
+		d, ok := diff[k]
+		// No change.
+		if !ok {
+			new[k] = v
+			continue
+		}
+		// Removed.
+		if isRemoved(d) {
+			continue
+		}
+		// Updated.
+		var newV interface{}
+		switch v := v.(type) {
```

Comment:
I think this switch would fit more cleanly at the top of the function (if you make Merge take interface{} as arguments).